"dlv attach ..." command fails with "The handle is invalid."

1. What version of Delve are you using (`dlv version`)?

I use commit adb1746

2. What version of Go are you using? (`go version`)?

go version go1.13.6 windows/amd64

3. What operating system and processor architecture are you using?

Windows 7

4. What did you do?

I run
```
go test -v -run=AttachDetach github.com/go-delve/delve/pkg/proc
```
command

5. What did you expect to see?

I expected test to pass.

6. What did you see instead?

The test fails, like this

```
=== RUN TestAttachDetach
--- FAIL: TestAttachDetach (5.58s)
    proc_test.go:111: failed assertion at proc_test.go:2845: Attach - The handle is invalid.
FAIL
FAIL github.com/go-delve/delve/pkg/proc 6.184s
FAIL
```

The `The handle is invalid.` (ERROR_INVALID_HANDLE) is returned from WaitForDebugEvent Windows API.

Reading WaitForDebugEvent documentation from https://docs.microsoft.com/en-us/windows/win32/api/debugapi/nf-debugapi-waitfordebugevent

> Only the thread that created the process being debugged can call WaitForDebugEven ...

What, I think, that means, is that DebugActiveProcess and WaitForDebugEvent APIs should be called by the same thread. But that is not the case at this moment.

The error does not happen every single time I run the test, but it happens often enough.

Alex

--- a/pkg/proc/native/proc_windows.go
+++ b/pkg/proc/native/proc_windows.go
@@ -152,8 +152,12 @@ func findExePath(pid int) (string, error) {
 
 // Attach to an existing process with the given PID.
 func Attach(pid int, _ []string) (*Process, error) {
-	// TODO: Probably should have SeDebugPrivilege before starting here.
-	err := _DebugActiveProcess(uint32(pid))
+	dbp := New(pid)
+	var err error
+	dbp.execPtraceFunc(func() {
+		// TODO: Probably should have SeDebugPrivilege before starting here.
+		err = _DebugActiveProcess(uint32(pid))
+	})
 	if err != nil {
 		return nil, err
 	}
@@ -161,7 +165,6 @@ func Attach(pid int, _ []string) (*Process, error) {
 	if err != nil {
 		return nil, err
 	}
-	dbp := New(pid)
 	if err = dbp.initialize(exepath, []string{}); err != nil {
 		dbp.Detach(true)
 		return nil, err

